ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -119,101 +116,170 @@ export const ConnectPage: React.FC<ConnectPageProps> = ({
+    (network: { caipChainId: CaipChainId }) =>
+      network.caipChainId === currentlySelectedNetworkChainId,
   );
 
   const defaultSelectedNetworkList = selectedTestNetwork
-    ? [...nonTestNetworks, selectedTestNetwork].map(({ chainId }) => chainId)
-    : nonTestNetworks.map(({ chainId }) => chainId);
+    ? [...nonTestNetworkConfigurations, selectedTestNetwork].map(
+        ({ caipChainId }) => caipChainId,
+      )
+    : nonTestNetworkConfigurations.map(({ caipChainId }) => caipChainId);
 
-  const allNetworksList = [...nonTestNetworks, ...testNetworks].map(
-    ({ chainId }) => chainId,
+  const defaultSelectedChainIds =
+    supportedRequestedCaipChainIds.length > 0
+      ? supportedRequestedCaipChainIds
+      : defaultSelectedNetworkList;
+
+  console.log({
+    currentlySelectedNetworkChainId,
+    selectedTestNetwork,
+    defaultSelectedNetworkList,
+    defaultSelectedChainIds,
+  });
+
+  const [selectedChainIds, setSelectedChainIds] = useState<CaipChainId[]>(
+    defaultSelectedChainIds as CaipChainId[],
   );
 
-  const supportedRequestedChainIds = requestedChainIds.filter((chainId) =>
-    allNetworksList.includes(chainId),
+  const allAccounts = useSelector(
+    getUpdatedAndSortedAccounts,
+  ) as MergedInternalAccount[];
+
+  const allAccountsWithCaipAccountId = allAccounts.map((account) => {
+    // I hope we can reliably use the first scope to determine the namespace
+    const { namespace, reference } = parseCaipChainId(account.scopes[0]);
+    return {
+      internalAccount: account,
+      caipAccountId:
+        `${namespace}:${reference}:${account.address}` as CaipAccountId,
+    };
+  });
+
+  const requestedNamespaces = getUniqueArrayItems(
+    [
+      ...Object.keys(requestedCaip25CaveatValue.requiredScopes),
+      ...Object.keys(requestedCaip25CaveatValue.optionalScopes),
+    ].map((scope) => {
+      const scopeString = scope as InternalScopeString;
+      // This will break if 'wallet' is passed in
+      if (scopeString === KnownCaipNamespace.Wallet) {
+        return scopeString;
+      }
+      const { namespace, reference } = parseCaipChainId(scopeString);
+      return namespace === KnownCaipNamespace.Wallet ? reference : namespace;
+    }),
   );
 
-  const defaultSelectedChainIds =
-    supportedRequestedChainIds.length > 0
-      ? supportedRequestedChainIds
-      : defaultSelectedNetworkList;
+  // all accounts that match the requested namespaces
+  const supportedAccountsForRequestedNamespaces =
+    allAccountsWithCaipAccountId.filter((account) => {
+      const {
+        chain: { namespace },
+      } = parseCaipAccountId(account.caipAccountId);
+      return requestedNamespaces.includes(namespace);
+    });
 
-  const [selectedChainIds, setSelectedChainIds] = useState(
-    defaultSelectedChainIds,
+  // all requested accounts that are found in the wallet
+  const supportedRequestedAccounts = Array.from(requestedCaipAccountIds).reduce(
```

Comment:
is `requestedCaipAccountIds` not an array already?

Reply:
checking this

Reply:
made the selector return an array instead of a set and then removed the Array.from 
https://github.com/MetaMask/metamask-extension/pull/31326/commits/c6e54164872e95ce5c772c9c4371759bf274de93